GoogleForCreators/web-stories-wp
# Workspace: Fix tooltip positioning in RTL

## Context

The tooltip positioning looked funky in RTL. Big thanks to @littlemilkstudio for helping me to figure this out.

## Summary

Fixes tooltip positioning in RTL in the dashboard and editor

## Relevant Technical Choices

The positioning is being calculated correctly in React. However, all the funky modal-placement logic for the tooltip doesn't get changed when we switch to RTL. Changes:

1. The horizontal positioning choices need to be flipped when in RTL. To do this, I made a wrapper in both the dashboard and editor that pulls the `isRTL` value from the config.
2. The arrow positioning needs to be recalculated when in RTL since it's based off of the distance between the elements. When in RTL we need to multiply by `-1` to make sure we position the arrow between the elements.
    - Needed to add a way for the tooltip to know the arrow needs to be mirrored across the y-axis. Added `isMirrored` prop to the tooltip to change the arrow position. Don't love it, but think it's the best way for now.

Honestly, this probably wouldn't be a problem if we weren't using Portals because we could calculate the tooltip position based off of its parent elements. The tooltip is rendered outside the React tree, so it has no awareness of the positioning of its parent elements.

## To-do

none

## User-facing changes

Most tooltips remain unchanged since they have centered horizontal positioning. There's only one tooltip which has any sort of horizontal offset. It is shown below:

|Before|After|
|--|--|
|![image](https://user-images.githubusercontent.com/22185279/112067722-275b8d80-8b2e-11eb-82bd-c0ed96f9b33f.png)|![image](https://user-images.githubusercontent.com/22185279/112067633-fc713980-8b2d-11eb-8b60-87954de7767a.png)|

![tooltip-fixed](https://user-images.githubusercontent.com/22185279/112067686-14e15400-8b2e-11eb-90c0-8e5c97a63ecd.gif)

## Testing Instructions

Unfortunately this will mean you have to test the tooltip positions in RTL for every tooltip in the application. The LTR tooltips didn't change, so that's nice.

### QA






This PR can be tested by following these steps:

1. Go to dashboard, switch to RTL
2. Hover the 'list view' icon 
3. Tooltip should not be cut off

### UAT






## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #6663 
Fixes #6755
